### PR TITLE
Add ipdb auth headers

### LINKED ISSUE #5: allow for flat chaining of request
Smth along the lines of:

```
let conn = driver.Connection(PATH);
conn
    .postTransaction(txSigned)
    .then(() => conn.getStatus(txSigned.id))
    .then((res) => console.log('Transaction status:', res.status));
```

### PR DESCRIPTION
**Pull request by TimDaub**
@diminator 

- included headers in `Connection` object
- combined files to make hacking easier
- rm HACKING.rst (What's the purpose anyways)
- Adjusted API.md and readme.md

Fixes #5 and #3

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -148,80 +148,73 @@ Returns **cc.Condition** Ed25519 Condition (that will need to wrapped in an Outp
 
 Returns **[object](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object)** Ed25519 Condition (that will need to wrapped in an Output)
 
-## getBlock
+## Connection
```
Comment (diminator) [inline API.md]: did you create these using documentation.js?

`npm install -g documentation`
`documentation build src/index.js -a public -f md > API.md  `

File: README.md [hunk 2]
```diff
@@ -75,13 +75,10 @@ const tx = driver.Transaction.makeCreateTransaction(
...
 
 // send it off to BigchainDB
-driver.Connection
-    .postTransaction(txSigned, API_PATH)
-    .then((res) => {
-        // request the status of the transaction
-        driver.Connection
-            .getStatus(txSigned.id, API_PATH)
-            .then((res) => console.log('Transaction status:', res.status))});
+let conn = driver.Connection(PATH);
```
Comment (diminator) [inline README.md]: I like this
remind me to upgrade:
- boilerplate
- kyber: examples & tutorials

File: src/connection/index.js [hunk 3]
```diff
@@ -1,9 +1,158 @@
...
-export listBlocks from './listBlocks';
-export listOutputs from './listOutputs';
-export listTransactions from './listTransactions';
-export listVotes from './listVotes';
-export pollStatusAndFetchTransaction from './pollStatusAndFetchTransaction';
-export postTransaction from './postTransaction';
+import request from '../request';
+
+
+class Connection {
```
Comment (diminator) [inline src/connection/index.js]: making this a class makes sense!

File: src/connection/index.js [hunk 4]
```diff
@@ -1,9 +1,158 @@
...
+            'blocks_detail': this.path + 'blocks/%(blockId)s',
+            'outputs': this.path + 'outputs',
+            'statuses': this.path + 'statuses',
+            'transactions': this.path + 'transactions',
+            'transactions_detail': this.path + 'transactions/%(txId)s',
+            'votes': this.path + 'votes'
+        }[endpoints];
+    }
+
+    req(path, options={}) {
```
Comment (diminator) [inline src/connection/index.js]: make this private?
`_req`

File: src/connection/index.js (hunk 3 above)
Comment (diminator) [inline src/connection/index.js]: should we do the same for `driver.Transaction`?

File: API.md (hunk 1 above)
Comment (TimDaub) [inline API.md]: damn!

I'll run it

File: README.md (hunk 2 above)
Comment (TimDaub) [inline README.md]: > remind me to upgrade:

I won't hahahhahaha *evillaughter*

File: src/connection/index.js (hunk 4 above)
Comment (TimDaub) [inline src/connection/index.js]: Yes

File: API.md (hunk 1 above)
Comment (TimDaub) [inline API.md]: Also thanks for providing the commands!

**Comment by TimDaub** [comment]
@diminator This should be ready now.

**Pull request merged**